pkg/scheduler/internal/cache/cache.go

```diff
@@ -569,15 +573,15 @@ func (cache *cacheImpl) RemovePod(pod *v1.Pod) error {
 		return fmt.Errorf("pod %v(%v) is not found in scheduler cache, so cannot be removed from it", key, klog.KObj(pod))
 	}
 	if currState.pod.Spec.NodeName != pod.Spec.NodeName {
-		klog.ErrorS(nil, "Pod was added to a different node than it was assumed", "podKey", key, "pod", klog.KObj(pod), "assumedNode", klog.KRef("", pod.Spec.NodeName), "currentNode", klog.KRef("", currState.pod.Spec.NodeName))
+		logger.Error(nil, "Pod was added to a different node than it was assumed", "podKey", key, "pod", klog.KObj(pod), "assumedNode", klog.KRef("", pod.Spec.NodeName), "currentNode", klog.KRef("", currState.pod.Spec.NodeName))
 		if pod.Spec.NodeName != "" {
 			// An empty NodeName is possible when the scheduler misses a Delete
 			// event and it gets the last known state from the informer cache.
-			klog.ErrorS(nil, "scheduler cache is corrupted and can badly affect scheduling decisions")
-			klog.FlushAndExit(klog.ExitFlushTimeout, 1)
+			logger.Error(nil, "scheduler cache is corrupted and can badly affect scheduling decisions")
+			os.Exit(1)
```

Comment:
It seems incompatible, but we do not have similar functions like `flushAndExit` under contextual logger, right?

Reply:
If we use klog here, then no need to update.

Reply:
@pohly  https://github.com/kubernetes/klog/pull/306 It looks like contextual logging now supports flush(klog.FlushAndExit), right?

Reply:
Yes. Use `logger.Error` and `klog.FlushAndExit` here.